File: Architecture.Web/ClientApp/src/app/Shared/Entity/ClientProfile/profFamilyInfo.ts
import { Data } from '@angular/router';
import { Auditable } from '../Common/auditable';

export class ProfFamilyInfo extends Auditable {
    public familyInfoId: number;
    public profileId: number;
    //public virtual ProfBasicInfo ProfBasicInfo { get; set; }
    public relationTypeId: number;
    //public virtual RelationType RelationType { get; set; }
    public relationType: string;
    public name: string;
    public surName: string;
    public taxCode: string;
    public dateOfBirth: Date;
    public placeOfBirth: string;
    public phoneNumber: string;
    public occupationTypeId: number;
    //public virtual OccupationType OccupationType { get; set; }
    public occupationType: string;
    public nationlityId: number;
    //public virtual Nationality Nationality { get; set; }
    public nationality: string;
    public previousNationality: string;
    public previousNationalityId: number;
    public residenceScopeId: number;
    //public virtual ResidenceScope ResidenceScope { get; set; }
    public residenceScope: string;
    public isDependent: boolean=false;
    public dependentPercentage: number;
    public isDisabled: boolean=false;
    public disabledPercentage: number;
    public yearlyIncome: number;
    public isAppliedForCitizenship: boolean=false;
    public applicationCode: string;
    public applicationDate: Date;
    public applicationCity: string;
    public applicationPlacedAddress: string;
    public applicationFileNumber: string;
    public applicationPlacedDate: Date;
}

File: Architecture.Core/Entities/ProfFamilyInfo.cs
using System;
using System.Collections.Generic;
using System.Text;
using Architecture.Core.Entities.Core;
using Architecture.Core.Entities.LU;

namespace Architecture.Core.Entities
{
   public class ProfFamilyInfo: Auditable
    {
        public int FamilyInfoId { get; set; }
        public int ProfileId { get; set; }
        public virtual ProfBasicInfo ProfBasicInfo { get; set; }
        public int? RelationTypeId { get; set; }
        public virtual RelationType RelationType { get; set; }
        public string Name { get; set; }
        public string SurName { get; set; }
        public string TaxCode { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string PlaceOfBirth { get; set; }
        public string PhoneNumber { get; set; }
        public int? OccupationTypeId { get; set; }
        public virtual OccupationType OccupationType { get; set; }
        public int? NationlityId { get; set; }
        public virtual Nationality Nationality { get; set; }
        public virtual Nationality PreviousNationality { get; set; }
        public int? PreviousNationalityId { get; set; }
        public int? ResidenceScopeId { get; set; }
        public virtual ResidenceScope ResidenceScope { get; set; }
        public bool IsDependent { get; set; } = false;
        public decimal DependentPercentage { get; set; }
        public bool IsDisabled { get; set; }
        public decimal DisabledPercentage { get; set; }
        public decimal YearlyIncome { get; set; }
        public bool IsAppliedForCitizenship { get; set; } = false;
        public string ApplicationCode { get; set; }
        public DateTime? ApplicationDate { get; set; }
        public string ApplicationCity { get; set; }
        public string ApplicationPlacedAddress { get; set; }
        public string ApplicationFileNumber { get; set; }
        public DateTime? ApplicationPlacedDate { get; set; }
        public int? GenderId { get; set; }
        public virtual Gender Gender { get; set; }
        public int? ProvinceId { get; set; }
        public virtual Province Province { get; set; }

    }
}
